test/BUILD.gn

```diff
@@ -452,9 +452,13 @@ test("brave_unit_tests") {
     sources += [
       "//brave/browser/ipfs/content_browser_client_helper_unittest.cc",
       "//brave/browser/ipfs/ipfs_host_resolver_unittest.cc",
+      "//brave/browser/ipfs/ipfs_navigation_throttle_unittest.cc",
```

Comment:
These also should be put in ipfs's own test target and be added into deps here instead.
I know there are a lot of files added directly into this file before, but we should avoid it from now on because it makes deps tracking much harder.
And since you're moving test files into a correct location, it would be good if we fix its BUILD.gn too.